crates/sui-framework-snapshot/tests/compatibility_tests.rs

```diff
@@ -67,4 +67,12 @@ mod compatibility_tests {
                 "The current framework differs the latest bytecode snapshot. Did you forget to upgrade protocol version?"
             );
     }
+
+    #[test]
+    fn check_no_dirty_manifest_commit() {
+        let snapshots = load_bytecode_snapshot_manifest();
+        for snapshot in snapshots.values() {
+            assert!(!snapshot.git_revision().contains("dirty"));
```

Comment:
Could you add a message here explaining what to do if it is dirty? (e.g. you probably don't want to run the snapshot generation on `main`, you want to run it on the release branch and pick it over to `main`)?